Sources/System/FileHelpers.swift

```diff
@@ -76,6 +76,76 @@ extension FileDescriptor {
     }
   }
 
+  /// Reads bytes at the current file offset into a buffer until the buffer is filled.
+  ///
+  /// - Parameters:
+  ///   - buffer: The region of memory to read into.
+  /// - Returns: The number of bytes that were read, equal to `buffer.count`.
+  ///
+  /// This method either reads until `buffer` is full, or throws an error if
+  /// only part of the buffer was filled.
+  ///
+  /// The <doc://com.apple.documentation/documentation/swift/unsafemutablerawbufferpointer/3019191-count> property of `buffer`
+  /// determines the number of bytes that are read into the buffer.
+  @_alwaysEmitIntoClient
+  @discardableResult
+  public func read(
+    filling buffer: UnsafeMutableRawBufferPointer
+  ) throws -> Int {
+    return try _read(filling: buffer).get()
+  }
+
+  /// Reads bytes at the current file offset into a buffer until the buffer is filled or until EOF is reached.
+  ///
+  /// - Parameters:
+  ///   - offset: The file offset where reading begins.
+  ///   - buffer: The region of memory to read into.
+  /// - Returns: The number of bytes that were read, at most `buffer.count`.
+  ///
+  /// This method either reads until `buffer` is full, or throws an error if
+  /// only part of the buffer was filled.
+  ///
+  /// The <doc://com.apple.documentation/documentation/swift/unsafemutablerawbufferpointer/3019191-count> property of `buffer`
+  /// determines the number of bytes that are read into the buffer.
+  @_alwaysEmitIntoClient
+  @discardableResult
+  public func read(
+    fromAbsoluteOffset offset: Int64,
+    filling buffer: UnsafeMutableRawBufferPointer
+  ) throws -> Int {
+    return try _read(fromAbsoluteOffset: offset, filling: buffer).get()
+  }
+
+  @usableFromInline
+  internal func _read(
```

Comment:
If we want to make this function opaque, then it'll need to come with availability, which means that the `read(filling:)` methods will need to have it too.

@milseman Is there a good reason `writeAll` is implemented using an opaque function?